clients/drcachesim/reader/reader.h

```diff
@@ -182,24 +189,21 @@ class reader_t : public std::iterator<std::input_iterator_tag, memref_t>,
     }
 
 protected:
-    // This reads the next entry from the stream of entries from all threads interleaved
-    // in timestamp order.
+    // This reads the next entry from the single stream of entries (or from the
+    // local queue if non-empty).  If it returns false it will set at_eof_ to distinguish
+    // end-of-file from an error.  It should call read_queue() first before reading
+    // a new entry from the input stream.
     virtual trace_entry_t *
     read_next_entry() = 0;
-    // This first checks the local queue before calling read_next_entry().
-    // in timestamp order.
+    // Returns and removes the entry (nullptr if none) from the local queue.
+    // This should be called by read_next_entry() prior to reading a new record
+    // from the input stream.
     virtual trace_entry_t *
-    read_next_entry_or_queue();
+    read_queue();
```

Comment:
s/read_queue/read_buffered_entry/

Or some other name that makes it clear what the queue is.